src/lib/connect/ws.ts

```diff
@@ -220,41 +224,27 @@ const browserStreamBuilder: StreamBuilder = (client, opts) => {
 	}
 
 	function onOpen() {
-		stream.setReadable(proxy)
-		stream.setWritable(proxy)
-		stream.emit('connect')
+		if (stream instanceof BufferedDuplex) {
+			stream.onSocketOpen()
+		}
 	}
 
 	function onClose() {
 		stream.end()
 		stream.destroy()
 	}
 
-	function onError(err: Event) {
-		stream.destroy(err)
+	function onError(err: ErrorEvent) {
+		stream.destroy(err as any)
 	}
 
 	function onMessage(event: MessageEvent) {
 		let { data } = event
 		if (data instanceof ArrayBuffer) data = Buffer.from(data)
-		else data = Buffer.from(data, 'utf8')
+		else data = Buffer.from(data as string, 'utf8')
 		proxy.push(data)
 	}
 
-	// this is to be enabled only if objectMode is false
-	function writev(chunks: any, cb: (err?: Error) => void) {
-		const buffers = new Array(chunks.length)
-		for (let i = 0; i < chunks.length; i++) {
-			if (typeof chunks[i].chunk === 'string') {
-				buffers[i] = Buffer.from(chunks[i], 'utf8')
```

Comment:
I also noticed this should have been `chunks[i].chunk`. Wondering how this cold ever be working before. Corrected [here](https://github.com/mqttjs/MQTT.js/pull/1732/files#diff-c807a1a5cf80c9f0c9fd037f1173b6e0beefe3672e805ff1392ef3be225858c7R15)